packages/devtools/lib/src/timeline/frames_bar_chart.dart

```diff
@@ -97,28 +97,78 @@ class PlotlyDivGraph extends CoreElement {
     }
   }
 
+  Selection currentSelection = null;
+
   void _plotlyClick(DataEvent data) {
     final int xPosition = data.points[0].x;
-    if (_frames.containsKey(xPosition)) {
-      final TimelineFrame timelineFrame = _frames[xPosition];
-      framesBarChart.setSelected(timelineFrame);
+
+    final List<int> pointNumbers = [];
+    final List<int> xValues = [];
+    final List<num> yValues = [];
+    final List<int> traces = [];
+
+    for (Point pt in data.points) {
+      // Don't allow selecting an already selected bar.
+      if (pt.curveNumber != FramesBarPlotly.gpuSelectTraceIndex &&
+          pt.curveNumber != FramesBarPlotly.cpuSelectTraceIndex) {
+        pointNumbers.add(pt.pointNumber);
+        traces.add(pt.curveNumber);
+        xValues.add(pt.x);
+        yValues.add(pt.y);
+      }
+    }
+
+    // Create selection once.
+    currentSelection ??= Selection(frameGraph, element);
+
+    // If this bar isn't currently selected then select the bar clicked.
+    if (xValues.length == 2 &&
```

Comment:
Why xValues.length == 2? Can you add a comment explaining this.

Reply:
It been changed to 
`newSelection.length
`
but two traces should exist, a point is always in 2 traces and only 2, otherwise there's a problem.  I'll add a comment it was strange before I think it's better now.